src/services/urlAssembly.ts

```diff
@@ -41,13 +41,17 @@ function assembleQueryParamValues(query: string, params: Record<string, Param>,
 
   return Array.from(search.entries()).reduce<QueryRecord>((url, [key, value]) => {
     const paramName = getParamName(value)
+    const isNotParam = !paramName
 
-    if (!paramName) {
+    if (isNotParam) {
       return { ...url, [key]: value }
     }
 
     const paramValue = setParamValue(paramValues[paramName], params[paramName])
-    if (isOptionalParamSyntax(value) && paramValues[paramName] === undefined && paramValue === '') {
+    const valueNotProvidedAndNoDefaultUsed = paramValues[paramName] === undefined && paramValue === ''
```

Comment:
I think this is unrelated to this PR but here's another empty string check that I'm curious why we need.

Reply:
I updated this bit of code because I found it hard to follow. Basically what it's doing is omitting from the search any params that were optional, not included in params, and do not have default values.

This prevents a `push('route-with-optional-query-params', {})` from giving you a URL like `/route?optional=&another=&third=`